stages/org.osbuild.ostree.deploy

```diff
@@ -112,39 +126,86 @@ def main(tree, inputs, options):
     ref = options["ref"]
     remote = options.get("remote")
 
-    # If provided an input then do the pull into the tree
-    if len(inputs) != 0:
-        source_repo, commits = ostree.parse_ostree_input(inputs)
-        target_repo = f"{tree}/ostree/repo"
-        for commit, data in commits.items():
-            loopref = data.get("ref", commit)
-            ostree.pull_local(source_repo, target_repo, remote, loopref)
-
-
-    if remote:
-        ref = f"{remote}:{ref}"
+    # distingush between ostree commit or container image input
+    input_type = ''
+    for key in inputs:
+       input_type = key
 
     kargs = []
 
-    if rootfs:
-        rootfs_id = make_fs_identifier(rootfs)
-        kargs += [f"--karg=root={rootfs_id}"]
-
-    for opt in kopts:
-        kargs += [f"--karg-append={opt}"]
+    if remote:
+      ref = f"{remote}:{ref}"
+
+    if len(inputs) == 0:
+    
+      if rootfs:
+          rootfs_id = make_fs_identifier(rootfs)
+          kargs += [f"--karg=root={rootfs_id}"]
+
+      for opt in kopts:
+          kargs += [f"--karg-append={opt}"]
+
+    elif len(inputs) == 1:
+
+        if input_type == 'images':
+          for opt in kopts:
+            kargs += [f"--karg={opt}"]
+
+        elif input_type == 'commits':
+          print('enter!!')
+          source_repo, commits = ostree.parse_ostree_input(inputs)
+          target_repo = f"{tree}/ostree/repo"
+          for commit, data in commits.items():
+              loopref = data.get("ref", commit)
+              ostree.pull_local(source_repo, target_repo, remote, loopref)
+    
+    # schema should catch the case in which there are more than one input but this adds a second layer of security
+    elif len(inputs) > 1:
+       raise ValueError("only one input accepted")
 
     with MountGuard() as mounter:
-        for mount in mounts:
-            path = mount.lstrip("/")
-            path = os.path.join(tree, path)
-            mounter.mount(path, path)
-
+      for mount in mounts:
+          path = mount.lstrip("/")
+          path = os.path.join(tree, path)
+          mounter.mount(path, path)
+          
+      if len(inputs) == 0:
         ostree.cli("admin", "deploy", ref,
-                   *kargs,
-                   sysroot=tree,
-                   os=osname)
-
-
+                *kargs,
+                sysroot=tree,
+                os=osname)
+      
+      elif input_type == 'images':
+        images = containers.parse_containers_input(inputs)
+        for i in inputs["images"]:
+            print(i)
+
+        for image in images.values():
+          with containers.container_source(image) as (image_name, image_source):
+            extra_args = []
+            imgref = f"{remote}:{image_source}"
```

Comment:
we probably shouldn't rely on `remote` here

```suggestion
            imgref = f"ostree-unverified-image:{image_source}"
```

Reply:
Do we want the ostree commit input case to also not rely on remote, just like how we changed the container image input case to not rely on remote (ie. `ref = f"fedora:{ref}"`)?

Reply:
Or maybe we should keep using remote since we still need to be able to accept an ostree commit created from a previous stage in the pipeline which might have a remote different from the `fedora` remote?